app/models/concerns/xml_message_object.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module XmlMessageObject
+  extend ActiveSupport::Concern
+
+  FORM_IDENTIFIER_PATTERN = /([^\/]+)\/(\d+\.\d+)\z/
+
+  included do
+    def pdf_transformation
+      return unless upvs_form&.xsl_fo
+      return unless unsigned_content
+      return unless xml?
+
+      template = Nokogiri::XSLT(upvs_form.xsl_fo)
+      fo_xml = template.transform(xml_unsigned_content)
+      fo_xml.encoding = 'UTF-8'
+
+      begin
+        xml_file = Tempfile.new("#{id}.xml")
+        xml_file.write(xml_unsigned_content)
+        xml_file.rewind
+
+        xsl_file = Tempfile.new("#{id}.fo.xsl")
+        xsl_file.write(upvs_form.xsl_fo.to_s)
+        xsl_file.rewind
+
+        pdf_file = Tempfile.new("#{id}.pdf")
+
+        system "fop -xml #{xml_file.path} -c #{Rails.root + 'config/fop.xconf'} -xsl #{xsl_file.path} -pdf #{pdf_file.path}"
+
+        pdf_file.rewind
+        pdf_file.read
+      ensure
+        xml_file.close
+        xsl_file.close
+        pdf_file.close
+
+        xml_file.unlink
+        xsl_file.unlink
+        pdf_file.unlink
+      end
+    end
+
+    def xml_unsigned_content
+      document = Nokogiri::XML(unsigned_content) do |config|
+        config.noblanks
+      end
+      document = Nokogiri::XML(document.xpath('*:XMLDataContainer/*:XMLData/*').to_xml) do |config|
+        config.noblanks
+      end if document.xpath('*:XMLDataContainer/*:XMLData').any?
+
+      document
+    end
+
+    def upvs_form
+      xml_document = xml_unsigned_content
+      posp_id, posp_version = xml_document.root.namespace&.href&.match(FORM_IDENTIFIER_PATTERN)&.captures
+
+      ::Upvs::Form.find_by(
+        identifier: posp_id,
+        version: posp_version
+      )
+    end
+
+    def xml?
+      if try(:is_signed)
```

Comment:
Toto je zvlastne. Moze to niekedy nemat metodu is_signed?

Reply:
Ano, ak to je `nested_message_object`, tam to nerozlisujeme. Iba pri `message_object`. Mozeme ale upravit.

Reply:
Skor mi dava zmysel, ze tu metodu proste ocakavame ze ma implementovanu. Ak tam nie je tak je to zle.

Reply:
Upravene, pridala som taku metodu.